Lib/argparse.py

```diff
@@ -1540,6 +1540,13 @@ def _get_positional_kwargs(self, dest, **kwargs):
             msg = _("'required' is an invalid argument for positionals")
             raise TypeError(msg)
 
+        # make sure 'store_true', 'store_false', or 'store_const' action
+        # is not specified
+        action = kwargs.get('action')
+        if action in ('store_true', 'store_false', 'store_const'):
+            msg = _("{action} is an invalid action for positionals", {'action': action})
```

Comment:
The argument of `_()` should be a string, not a tuple. I afraid that your tests are passed because a wrong TypeError is raised. You should test also the error message to be sure that this is the correct error.

Reply:
And these error messages should not be internationalized, as they are programming errors, not user input errors.

Reply:
Thank you for the review and comments. I will get back to this PR later this week.

Reply:
Actually, I already created a PR with more general solution: #124839.